# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1358297538 18000
#      Tue Jan 15 19:52:18 2013 -0500
# Branch classdef
# Node ID db6371b97fed5c44badbca3b146e0d38b54a3801
# Parent  d8553705f8f0cc9bd91fc4ae37ec9582388a274b
Fix bug in test/classdef/foo_payment.m.

* test/classdef/foo_payment.m: Using struct() to initialize object
  is not valid syntax.

diff --git a/test/classdef/foo_payment.m b/test/classdef/foo_payment.m
--- a/test/classdef/foo_payment.m
+++ b/test/classdef/foo_payment.m
@@ -1,23 +1,21 @@
 classdef foo_payment
   properties
     rate;
     term;
     principle;
   end
   methods
     function obj = foo_payment (r, t, p)
-      if (nargin == 0)
-        obj = struct ('rate', 0, 'term', 0, 'principle', 0);
-      elseif (nargin == 3)
+      if (nargin == 3)
         obj.rate = r;
         obj.term = t;
         obj.principle = p;
-      else
+      elseif (nargin ~= 0)
         error ('foo_payment:SyntaxError', ...
                'foo_payment: Invalid syntax')
       end
     end
     function amt = amount (obj)
       i = obj.rate / (12 * 100);
       if (i == 0 && obj.term == 0)
         amt = obj.principle;
